src/sympc/protocol/aby3/aby3.py

```diff
@@ -204,3 +207,201 @@ def _get_truncation_pair(x: MPCTensor, session: Session) -> Tuple[MPCTensor]:
+
+        input_rst = []
+        if bitwise:
+            for i in range(ring_bits):
+                input_rst.append(x.bit_extraction(i))
+        else:
+            input_rst.append(x)
+
+        for share in input_rst:
+            shares = [[zero.clone(), zero.clone()] for i in range(nr_parties)]
+
+            shares[rank][0] = share.shares[0].clone().type(tensor_type)
+
+            shares[(rank + 1) % nr_parties][1] = (
+                share.shares[1].clone().type(tensor_type)
+            )
+
+            rst_sh = []
+            for i in range(nr_parties):
+                rst = x.clone()
+                rst.shares = shares[i]
+                rst.ring_size = ring_size
+                rst_sh.append(rst)
+
+            share_lst.append(rst_sh)
+
+        return share_lst
+
+    @staticmethod
+    def bit_injection(x: MPCTensor, session: Session, ring_size: int) -> MPCTensor:
+        """Perfom ABY3 bit injection for conversion of binary share to arithmetic share.
+
+        Args:
+            x (MPCTensor) : MPCTensor with shares of bit.
+            session (Session): session the share belongs to.
+            ring_size (int) : Ring size of arithmetic share to convert.
+
+        Returns:
+            arith_share (MPCTensor): Arithmetic shares of bit in input ring size.
+
+        Raises:
+            ValueError: If input tensor is not binary shared.
+        """
+        inp_ring = int(x.share_ptrs[0].get_ring_size().get_copy())  # input ring_size
+        if inp_ring != 2:
+            raise ValueError("Bit injection works only for binary rings")
+
+        args = [[share, str(ring_size)] for share in x.share_ptrs]
+
+        decompose = parallel_execution(ABY3.local_decomposition, session.parties)(args)
+
+        x1_sh, x2_sh, x3_sh = zip(*map(lambda x: x[0], decompose))
+
+        x1_sh = [ptr.resolve_pointer_type() for ptr in x1_sh]
+        x2_sh = [ptr.resolve_pointer_type() for ptr in x2_sh]
+        x3_sh = [ptr.resolve_pointer_type() for ptr in x3_sh]
+
+        x1 = MPCTensor(shares=x1_sh, session=session, shape=x.shape)
+        x2 = MPCTensor(shares=x2_sh, session=session, shape=x.shape)
+        x3 = MPCTensor(shares=x3_sh, session=session, shape=x.shape)
+
+        arith_share = x1 ^ x2 ^ x3
+
+        return arith_share
+
+    @staticmethod
+    def full_adder(
+        a: List[MPCTensor], b: List[MPCTensor], session: Session
+    ) -> List[MPCTensor]:
+        """Perfom bit addition on MPCTensors using a full adder.
+
+        Args:
+            a (List[MPCTensor]): MPCTensor with shares of bit.
+            b (List[MPCTensor]): MPCTensor with shares of bit.
+            session (Session): session the share belongs to.
+
+        Returns:
+            result (List[MPCTensor]): Result of the operation.
+        """
+        ring_size = session.ring_size
+        ring_bits = get_nr_bits(ring_size)
+        c = [0 for idx in range(ring_bits + 1)]  # carry bits of addition.
```

Comment:
NITPICK: `c = [0 for _ in range(ring_bits + 1)]` (no need for idx)

Reply:
Sure George :+1:

Reply:
Done